core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+      showMessage('suggestions notifications via email are enabled.');
+    } else {
+      throw new Error('suggestions notifications via email are disabled.');
+    }
+  }
+
+  /**
+   * This funciton helps in deleting the exploration successfully.
+   */
+  async deleteExploration(): Promise<void> {
+    await this.page.waitForSelector(
+      `${deleteExplorationButton}:not([disabled]`
+    );
+    await this.clickOn(deleteExplorationButton);
+    await this.clickOn(deleteConfirmButton);
+  }
+
+  /**
+   * This function helps in verifying, if exploration is
+   * deleted successfully or not.
+   */
+  async expectExplorationToBeDeletedSuccessfullyFromCreatorDashboard(): Promise<void> {
+    try {
+      await this.goto(explorationUrlAfterPublished);
+      throw new Error('Exploration is not deleted successfully.');
+    } catch (error) {
+      showMessage('Exploration is successfully deleted.');
+    }
+  }
+
+  /**
+   * This function helps in drafting the exploration.
+   */
+  async saveDraftExploration(): Promise<void> {
+    await this.page.keyboard.press('Tab');
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileSaveDraftButton);
+    } else {
+      await this.clickOn(saveDraftButton);
+    }
+
+    await this.clickOn(commitMessage);
+    await this.type(commitMessage, 'Testing Testing');
+    await this.clickOn('.e2e-test-save-draft-button');
+  }
+
+  /**
+   * This function checks whether changes has been drafted or not.
+   */
+  async expectExplorationToBeDraftedSuccessfully(): Promise<void> {
+    await this.page.waitForSelector('#tutorialSaveButton');
+    const button = await this.page.$('#tutorialSaveButton');
+
+    if (!button) {
+      throw new Error('Save button not found.');
+    }
+
+    const isDraftButtonDisabled = await button.evaluate(
+      button => (button as HTMLButtonElement).disabled
+    );
+
+    if (isDraftButtonDisabled) {
+      showMessage('Changes have been successfully drafted.');
+    } else {
+      throw new Error('Failed to draft changes.');
+    }
+  }
+
+  /**
+   * This function helps in publishing the exploration
+   */
+  async publishExploration(): Promise<void> {
+    await this.saveDraftExploration();
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(publishButtonDropDown);
+      await this.clickOn(mobilePublishButton);
+    } else {
+      await this.page.waitForSelector(`${publishButton}:not([disabled])`);
+      await this.clickOn(publishButton);
+    }
+    await this.page.waitForSelector(`${publishConfirmButton}:not([disabled])`);
+    await this.clickOn(publishConfirmButton);
+    await this.page.waitForSelector(closePublishedPopUp, {visible: true});
+    await this.clickOn(closePublishedPopUp);
+
+    explorationUrlAfterPublished = await this.page.url();
+  }
+
+  /**
+   * This function checks whether the exploration
+   * is published successfully or not.
+   */
+  async expectInteractionOnCreatorDashboard(): Promise<void> {
```

Comment:
I have no idea what this means. An interaction is a question type, and question types do not appear on the creator dashboard.

If you want to check whether something is on the creator dashboard, go to /creator-dashboard, get a list of names shown there, and check whether the one you're looking for is contained in that list. See above.

Reply:
Changed to `expectExplorationToBePresentOnCreatorDashboard`, 
Similiar logic as `expectExplorationToNotAppearInCreatorDashboard`